src/algorithms/models/graph.ts

```diff
@@ -33,11 +33,22 @@ export default class Graph {
      */
     getVertex(id: number): Vertex {
         if (!this.vertices.has(id)) {
-            throw new Error('Vertex ID not found!');
+            throw new Error('Vertex ' + id + ' not found!');
         }
         return this.vertices.get(id);
     }
 
+    /**
+     * Get function which returns a set of all vertex IDs contained within the graph
+     */
+    getVertexIDs(): Set<number> {
+        const idSet = new Set<number>();
+        for (const vertex of this.vertices.values()) {
```

Comment:
A lot of copying over here... I'd rather access the vertices directly and not modify them (read-only) - I suggest my solution in later comment